src/litserve/server.py

```diff
@@ -19,15 +20,59 @@
 LIT_SERVER_API_KEY = os.environ.get("LIT_SERVER_API_KEY")
 
 
-def inference_worker(lit_api, device, worker_id, request_queue, request_buffer):
-    lit_api.setup(device=device)
+def get_batch_from_uid(uids, lit_api, request_buffer):
+    batches = []
+    for uid in uids:
+        try:
+            x_enc, pipe_s = request_buffer.pop(uid)
+        except KeyError:
+            continue
+        x = lit_api.decode_request(x_enc)
+        batches.append((x, pipe_s))
+    return batches
+
+
+def collate_requests(lit_api, request_queue: Queue, request_buffer, max_batch_size, batch_timeout):
+    uids = []
+    entered_at = time.time()
+    while (batch_timeout - (time.time() - entered_at) > 0) and len(uids) <= max_batch_size:
+        try:
+            uid = request_queue.get_nowait()
+            uids.append(uid)
+        except (Empty, ValueError):
```

Comment:
Should we add a `time.sleep(0.001)` to avoid looping too aggressively?

Reply:
how about `uid = request_queue.get(timeout=0.001)z

Reply:
Ok let's do that